### PR TITLE
Add item port

### PR DESCRIPTION
**Pull request by phoenixbox**
#### What's this PR do?
Moves the add item functionality into the drawer design pattern we followed for the item detail page
Refactors multiple subitem components used across item card and add item into two `./component/subitems&subitem`

#### Where should the reviewer start?
`pages/add-item`

#### How should this be manually tested?
Open the app and visit `/add-item`
Under a story confirm that you can add subitems and remove them
Hit create and see that the subitems get batch created after

#### Background Context
I updated the event change emission signature on the product store to accept a `type && record` param so that the view component could look for passed arguments within its `_onChange` function.
Open to other suggestions/patterns. I failed to be able to find a way to return a reference to the newly created item via the Dispatcher otherwise.

I made the `subitems` component more generic so that its less context aware and can be repurposed across the

* item-card
* add-item
* item-detail on mobile

No attachment support yet.

#### Screenshots (if appropriate)
image

image

### CODE REVIEW

File: app/actions/item-actions.js [hunk 1]
```diff
@@ -19,18 +22,51 @@ let ItemActions = {
...
+    Promise.map(data, (newItem) => {
+      let newItemData = this.itemForType(type, newItem, parent.number);
+      let item = product.createItem(newItemData, { silent: true });
+
+      item.save();
+
+      return item;
+    }).then(function(items) {
+      _.each(items, (item) => {
+        AppDispatcher.dispatch({
```
Comment (wookiehangover) [inline app/actions/item-actions.js]: Can we dispatch multiple events like this? I thought that you weren't supposed to call `dispatch` more than once inside the same frame

File: app/views/components/header.js [hunk 2]
```diff
@@ -2,7 +2,9 @@ import _ from 'lodash';
 import React from 'react/addons';
 import classNames from "classnames";
 import Gravatar from './gravatar';
-import AddItemModal from './add-item-modal';
+
+// import AddItemModal from './add-item-modal';
```
Comment (wookiehangover) [inline app/views/components/header.js]: let's just delete the dead codez

Comment (phoenixbox) [inline app/views/components/header.js]: Im sentimental :stuck_out_tongue:

File: app/actions/item-actions.js (hunk 1 above)
Comment (phoenixbox) [inline app/actions/item-actions.js]: So I havent seen any errors off the back of this - but I can just dispatch one instead

**Comment by phoenixbox** [comment]
@wookiehangover caught an incorrect reference to the parent item which was breaking things

Should be good to push to a staging branch now :+1:

**Pull request merged**